utils/src/server/rpc_module.rs

```diff
@@ -143,20 +217,21 @@ impl RpcModule {
 		Ok(SubscriptionSink { method: subscribe_method_name, subscribers })
 	}
 
-	/// Convert a module into methods.
```

Comment:
right, this doesn't make sense anymore because `Methods` and `AsyncMethods` are two different types, but why not `-> (Methods, AsyncMethods)`?

Reply:
I thought that keeping the resolving logic in `RpcModule` can lead to a bit more simple code. Like, the lookup of method existence in both sync and methods collections is unified under the `method_type`.

All the methods that should not be called in the context where this removed method would've been called are taking `&mut self` reference, thus we won't be able to call them as long as we keep the object immutable.

Or is there a reasoning that I don't see?

Reply:
I think it might be issues with #350 that's all I can think of.

Reply:
I've created a `MethodsHolder` structure which contains both `Methods` and `AsyncMethods` and has a convenient interface as a midground.